### PR TITLE
feat: implement success and failure pages

### PR DESCRIPTION
**Pull request by Zied-Dahmani**
## Scope

[Ticket](https://github.com/hypha-dao/hypha-wallet/issues/350)

Refactored the `Success` and `Failure` pages to be reusable for handling various outcomes, such as `successfully published proposals` or `failed publications`.

Before            |  After
:-------------------------:|:-------------------------:
 |

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1097,7 +1097,7 @@ PODS:
     - PromisesSwift (~> 2.1)
   - FirebaseSharedSwift (10.29.0)
   - Flutter (1.0.0)
-  - flutter_keyboard_visibility (0.0.1):
+  - flutter_keyboard_visibility_temp_fork (0.0.1):
     - Flutter
```
Comment (n13) [inline ios/Podfile.lock:1101]: Why do we need this "temp fork" package? Does it solve any issues we had?

**Review by n13 — CHANGES_REQUESTED** [review]
Please add screenshots with before/after like @nbetsaif always does, for visual changes! 

also see comments

File: lib/design/status/success_page.dart [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
 import 'package:hypha_wallet/ui/bottom_navigation/hypha_bottom_navigation.dart';
+import 'package:hypha_wallet/ui/proposals/details/proposal_details_page.dart';
 
-enum SignSuccessTransactionType {
-  approved('Approved');
+enum SuccessType {
+  approved('Approved'),
+  published('published');
 
-  const SignSuccessTransactionType(this.value);
```
Comment (n13) [inline lib/design/status/success_page.dart:14]: Why have these pages be renamed? They're still about signing, they should keep the same, more descriptive name, they have before.

File: lib/design/status/failure_page.dart [hunk 3]
```diff
@@ -10,10 +10,12 @@ import 'package:hypha_wallet/design/hypha_colors.dart';
 import 'package:hypha_wallet/design/themes/extensions/theme_extension_provider.dart';
 import 'package:hypha_wallet/ui/bottom_navigation/hypha_bottom_navigation.dart';
 
-class SignTransactionFailedPage extends StatelessWidget {
+class FailurePage extends StatelessWidget {
```
Comment (n13) [inline lib/design/status/failure_page.dart:13]: Why has this class / file been renamed?

File: lib/design/status/success_page.dart [hunk 4]
```diff
@@ -1,39 +1,35 @@
...
 
   final String value;
 
-  Color get iconBackgroundColor {
-    switch (this) {
-      case SignSuccessTransactionType.approved:
-        return HyphaColors.success;
-    }
-  }
+  Color get iconBorderColor => HyphaColors.offWhite;
```
Comment (n13) [inline lib/design/status/success_page.dart:22]: This seems like a breaking change - it should still use success as color?

File: lib/design/status/success_page.dart [hunk 5]
```diff
@@ -61,7 +57,7 @@ class SignTransactionSuccessPage extends StatelessWidget {
         bottomNavigationBar: HyphaSafeBottomNavigationBar(
           child: HyphaAppButton(
             onPressed: () {
-              Get.offAll(() => const HyphaBottomNavigation());
+              Get.Get.offAll(() => const HyphaBottomNavigation());
```
Comment (n13) [inline lib/design/status/success_page.dart:60]: Why Get.Get... ?

File: lib/design/status/success_page.dart [hunk 6]
```diff
@@ -82,14 +78,34 @@ class SignTransactionSuccessPage extends StatelessWidget {
                   Padding(padding: const EdgeInsets.only(left: 45, right: 45), child: successText),
                   const SizedBox(height: 46),
                   Container(
-                    padding: const EdgeInsets.all(16),
+                    padding: const EdgeInsets.all(12),
```
Comment (n13) [inline lib/design/status/success_page.dart:81]: Need visual before after screenshots...

File: lib/ui/proposals/creation/interactor/proposal_creation_state.dart [hunk 7]
```diff
@@ -3,9 +3,8 @@ part of 'proposal_creation_bloc.dart';
 @freezed
 class ProposalCreationState with _$ProposalCreationState {
   const factory ProposalCreationState({
-    @Default(PageState.initial) PageState pageState,
-    ProposalCreationModel? proposal,
```
Comment (n13) [inline lib/ui/proposals/creation/interactor/proposal_creation_state.dart:7]: don't need PageState anymore?

File: lib/design/status/failure_page.dart (hunk 3 above)
Comment (Zied-Dahmani) [reply lib/design/status/failure_page.dart:13]: I initially thought I would use it in a feature unrelated to signing transactions, but I realized that's not the case, so I'll revert it.

**Comment by Zied-Dahmani** [comment]
> Please add screenshots with before/after like @nbetsaif always does, for visual changes!

I've added the screenshots as you requested. 
(I typically include them as well, but I thought they weren't necessary this time since the changes are minor.)

**Comment by n13** [comment]
> > Please add screenshots with before/after like @nbetsaif always does, for visual changes!
> 
> I've added the screenshots as you requested. (I typically include them as well, but I thought they weren't necessary this time since the changes are minor.)

They may be minor but I  don't know that until I see the screenshots...

File: lib/ui/sign_transaction/success/sign_transaction_success_page.dart [hunk 8]
```diff
@@ -3,37 +3,32 @@ import 'package:get/get.dart';
...
 
   final String value;
 
-  Color get iconBackgroundColor {
-    switch (this) {
-      case SignSuccessTransactionType.approved:
-        return HyphaColors.success;
-    }
-  }
+  Color get iconBorderColor => HyphaColors.offWhite;
```
Comment (n13) [inline lib/ui/sign_transaction/success/sign_transaction_success_page.dart:21]: Is there a reason you changed the color from HyphaColors.success to HyphaColors.offWhite?

If not, please revert to HyphaColors.success

**Review by n13 — CHANGES_REQUESTED** [review]
Just 1 minor thing then it's good to go

File: lib/ui/sign_transaction/success/sign_transaction_success_page.dart (hunk 8 above)
Comment (Zied-Dahmani) [reply lib/ui/sign_transaction/success/sign_transaction_success_page.dart:21]: As I mentioned, Luigi updated the UI. The success color was used for the background, but now I'm using off-white for the circle border.

**Review by n13 — APPROVED** [review]
All good thank you!

**Pull request merged**